scripts/print_run_metadata.py

```diff
@@ -0,0 +1,122 @@
+    else:
+        image_mapping_settings['use_peak_times'] = False
+
+    # Load options related to data loading
+    data_loader_mode = "train"
+    
+    # Create data processor
+    if apply_processing:
+        data_processor = DataProcessor(
+                image_mapper=ImageMapper(**image_mapping_settings),
+                **data_processing_settings)
+    else: data_processor = None
+
+    # Define data loading functions
+    if data_format == 'HDF5':
+        data_loader = HDF5DataLoader(
+                data_files,
+                mode=data_loader_mode,
+                data_processor=data_processor,
+                image_mapper=ImageMapper(**image_mapping_settings),
+                **data_loading_settings)
+
+    metadata = data_loader.get_metadata()
+
+    out = open(args.out_file,"w") if args.out_file else sys.stdout
+    
+    print("Tel {} ({} mode)\n".format(config['Data']['Loading']['selected_tel_type'], config['Data']['Loading']['example_type']), file=out)
+    
+    for i in metadata:
+        print("{}: {}\n".format(i,metadata[i]), file=out)
+
+    num_events_after_cuts_by_class_name = metadata['num_events_after_cuts_by_class_name']
+    total_num_events = sum(num_events_after_cuts_by_class_name.values())
+    print("\n{} total events.".format(total_num_events), file=out)
+    print("Num events by particle_id:", file=out)
+    for class_name in num_events_after_cuts_by_class_name:
+        print("{}: {} ({}%)".format( 
+                class_name, 
+                num_events_after_cuts_by_class_name[class_name], 
+                100 * float(num_events_after_cuts_by_class_name[class_name])/total_num_events),
+                file=out
+                )
+
+    num_images_after_cuts_by_class_name = metadata['num_images_after_cuts_by_class_name']
+    total_num_images = sum(num_images_after_cuts_by_class_name.values())
+    print("\n{} total images.".format(total_num_images), file=out)
+    print("Num images by particle_id:", file=out)
+    for class_name in num_images_after_cuts_by_class_name:
+        print("{}: {} ({}%)".format( 
+                class_name, 
+                num_images_after_cuts_by_class_name[class_name], 
+                100 * float(num_images_after_cuts_by_class_name[class_name])/total_num_images),
+                file=out
+                )
+            
+    num_val_examples_by_class_name = metadata['num_val_examples_by_class_name']
+    total_num_events = sum(num_val_examples_by_class_name.values())
+    print("\n{} total val examples.".format(total_num_events), file=out)
+    print("Num val examples by particle_id:", file=out)
+    for class_name in num_val_examples_by_class_name:
+        print("{}: {} ({}%)".format( 
+                class_name, 
+                num_val_examples_by_class_name[class_name], 
+                100 * float(num_val_examples_by_class_name[class_name])/total_num_events),
+                file=out
+                )
+    
```

Comment:
It says "no newline at end of file"

Reply:
Thanks, I have pushed a new commit. On my version of python it runs smoothly, let me know if it keeps giving you trouble.